### DISCUSSION TITLE
Is reanimated v1 more performant than v2??

### DISCUSSION THREAD

**Discussion by alexfov** [software-mansion/react-native-reanimated#1632] [General]
Hello guys.  I just tested v1 vs v2 and I am a bit curious. In test 40 objects with rotate transformations for each. v2 shows 40-50 fps, when v1 shows easy 60fps. Device - redmi note 7.

```js
  //v1
  const rotation = useValue(0);
  const rotate = concat(
    interpolateNode(rotation, { inputRange: [0, 1], outputRange: [-15, 15] }),
    "deg"
  );
  //loop from redash
  useCode(() => [set(rotation, loop({ duration: 10000, boomerang: true }))], []);

  //v2
  const rotate = useSharedValue(-15);
  rotate.value = withRepeat(withTiming(15, { duration: 10000 }), -1, true);
  const style = useAnimatedStyle(() => {
    return {
      transform: [{ rotate: rotate.value + "deg" }],
    };
  });
```

Ofcourse v2 is much convinient, but such fps losses... What do you guys think about this? Have you tried to compare performance?

**Comment by mrousavy** [comment]
Currently, v2 is "more performant" than v1 in cases where there are not a lot of Shared Values. Once you start animating a lot of things (multiple nodes in v1, shared values in v2), v1 is probably faster. This will likely change in the future though.

**Reply by alexfov** [reply]
Thank you for your feedback!

**Reply by nandorojo** [reply]
Yeah, I've noticed this too. Is the solution to avoid animating too much with v2 right now? Many key use-cases, such as lists, are hard to do at all.

**Reply by sergeymorkovkin** [reply]
@mrousavy Marc, and could you please elaborate a bit more on your conclusion? Do you mean reanimated v2 is slower than v1 because of what exactly? What do you mean by "this will likely change in the future"? I've seen that v2 has animation functions are implemented purely in JS. Did you mean this is slowing it down? Or maybe something else? Thanks in advance for clarification.

**Reply by whalemare** [reply]
Still waiting solution, because fps drop is still persist in 2.1.0

**Reply by lightrow** [reply]
fps issues and also bugs still present in v2 that force us to fall back to v1 in some cases.

**Reply by lightrow** [reply]
sigh https://github.com/software-mansion/react-native-reanimated/issues/3296
